File: 중국인의 나머지 정리/백준_6064_카잉 달력/카잉달력.cpp
#include <iostream>
#include <cmath>
using namespace std;

int GCD(int a, int b)
{
    int left = 0;
    while (1)
    {
        left = a % b;
        if (left == 0)
            break;
        a = b;
        b = left;
    }
    return b;
}

int LCD(int a, int b)
{
    int gcd = GCD(a, b);
    return (a / gcd * b / gcd * gcd);
}

int main()
{

    int _case;
    scanf("%d", &_case);
    int M, N, x, y;
    while (_case--)
    {
        scanf("%d %d %d %d", &M, &N, &x, &y);

        int i = x;
        bool is = false;
        int lcd = LCD(M, N);

        if (N == y)
            y = 0;
        while (i <= lcd)
        {
            if (i % N == y)
            {
                is = true;
                break;
            }
            i += M;
        }

        if (is)
            printf("%d\n", i);
        else
            printf("-1\n");
    }
}

File: 중국인의 나머지 정리/백준_6064_카잉 달력/카잉달력.kt
import java.util.*

fun main(args: Array<String>) {
    val s = Scanner(System.`in`)
    var _case = s.nextInt()
    while (_case-- > 0) {
        var M = s.nextInt()
        var N = s.nextInt()
        var x = s.nextInt()
        var y = s.nextInt()

        var i = x
        var iss = false

        var a = M
        var b = N
        var left = 0
        while (true) {
            left = a % b
            if (left == 0) break
            a = b
            b = left
        }

        var gcd = b
        var lcd = (M / gcd * N / gcd * gcd)

        if (N == y) y = 0
        while (i <= lcd) {
            if (i % N == y) {
                iss = true
                break
            }
            i += M
        }
        if (iss) println(i) else println(-1)
    }
}
